internal/providers/github/clients/app_test.go

```diff
@@ -39,6 +39,78 @@ import (
 	minderv1 "github.com/stacklok/minder/pkg/api/protobuf/go/minder/v1"
 )
 
+func TestParseV1AppConfig(t *testing.T) {
+	t.Parallel()
+
+	scenarios := []struct {
+		name       string
+		config     json.RawMessage
+		error      string
+		ghEvalFn   func(*minderv1.GitHubAppProviderConfig)
+		provEvalFn func(*minderv1.ProviderConfig)
+	}{
+		{
+			name:   "valid app config",
+			config: json.RawMessage(`{ "github-app": { "endpoint": "https://api.github.com" } }`),
+			ghEvalFn: func(ghConfig *minderv1.GitHubAppProviderConfig) {
+				assert.Equal(t, "https://api.github.com", ghConfig.Endpoint)
+			},
+			provEvalFn: func(providerConfig *minderv1.ProviderConfig) {
+				assert.Nil(t, providerConfig)
+			},
+		},
```

Comment:
I don't know if this was intended, but these functions capture `t` from the outer scope, making them report failures on `TestParseV1AppConfig` rather than e.g. `TestParseV1AppConfig/valid_app_and_provider_config`.

In case this was not the intended behaviour, they would have to receive it as an argument.

```suggestion
	scenarios := []struct {
		name       string
		config     json.RawMessage
		error      string
		ghEvalFn   func(*testing.T, *minderv1.GitHubAppProviderConfig)
		provEvalFn func(*testing.T, *minderv1.ProviderConfig)
	}{
		{
			name:   "valid app config",
			config: json.RawMessage(`{ "github-app": { "endpoint": "https://api.github.com" } }`),
			ghEvalFn: func(t *testing.T, ghConfig *minderv1.GitHubAppProviderConfig) {
				assert.Equal(t, "https://api.github.com", ghConfig.Endpoint)
			},
			provEvalFn: func(t *testing.T, providerConfig *minderv1.ProviderConfig) {
				assert.Nil(t, providerConfig)
			},
		},
```

Reply:
not intended, thank you. that was a late-evening refactoring gone wrong.